### ISSUE TITLE
Cannot enable TOTP Second-factor Auth

### ISSUE THREAD

**Issue by erudisch** [nextcloud/twofactor_totp#166]
According to the title I'm not able to enable the TOTP Second-factor Auth, because there is no enable checkbox (see attachment).

![nextcloud](https://cloud.githubusercontent.com/assets/4243192/25066480/e3e04010-2226-11e7-9c0d-ed13146e5810.png)

Any suggestions?

Best regards
Erik

**Comment by Dennis1993** [comment]
Do you use an adblocker which is removing the button/Checkbox?

**Comment by erudisch** [comment]
No, use it with pure epiphany without any plugins. Despite that, other checkboxes are working (see screenshot).

![nextcloud2](https://cloud.githubusercontent.com/assets/4243192/25066828/09873106-2232-11e7-8a5b-1af69e5b1c46.png)

**Comment by ChristophWurst** [comment]
How did you install the app?

**Comment by ChristophWurst** [comment]
I assume you've either git cloned it or downloaded the sources from github. Use a release from the app store instead. Thx

**Comment by erudisch** [comment]
I've installed the app using the admin account via the web interface of nextcloud from inside the "apps" page (.../index.php/settings/apps). I've also tried a version from github, this also doesn't work either as you said.

Maybe there are some security settings I've setted up (on the web server side) which are incompatible with this app. I will try it without any security settings activated and see if it works or not and respond to you later.

Or do you have any other ideas?

Kind Regards
Erik

**Comment by erudisch** [comment]
After investigating the apps folder of nextcloud, i found out that there is just an app called "twofactor_backupcodes" but no app called "twofactor_totp". Could this be the problem?

Regards
Erik

**Comment by erudisch** [comment]
OK, after comparing the sources in github and the installed version in my apps folder I think that there are two different apps. The one that is installed is for backup purpose only and the one on github (which is linked from the nextcloud app store) is the actual 2fa app.

When I install the github app manually, parallel to the app store app, nextcloud ask me to update it, so it believes they are the same apps, but one is called "twofactor_backupcodes" and the other one (from github) is called "twofactor_totp".

So maybe there is a problem in publishing the app to the nextcoud store?

Regards
Erik

**Comment by rigrig** [comment]
No, those really are two separate apps:
Nextcloud provides the general 2fa framework, and comes bundled with the backup codes app.
This app enables 2fa using TOTP.

**Comment by rigrig** [comment]
There is some javascript involved with showing the settings, maybe there is some problem with the app in epiphany?
Could you try another browser? (Or I'll try it in epiphany myself when I get home tonight)

**Comment by erudisch** [comment]
I've also tried it with Firefox and Chromuim - no Plugins active, same faulty result. I'll check my CSP settings of my web server this evening, may be they are to strict...

By the way, do you habe any recommendations for security settings of an Apache/Nginx hosting nextcloud?

Regards
Erik

Am 16. April 2017 13:04:47 MESZ schrieb Richard de Boer <notifications@github.com>:
>There is some javascript involved with showing the settings, maybe
>there is some problem with the app in epiphany?
>Could you try another browser? (Or I'll try it in epiphany myself when
>I get home tonight)
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/nextcloud/twofactor_totp/issues/166#issuecomment-294346031

-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

**Comment by ChristophWurst** [comment]
> By the way, do you habe any recommendations for security settings of an Apache/Nginx hosting nextcloud?

Check out https://docs.nextcloud.com/server/11/admin_manual/. You can also always ask for help on our forums at http://help.nextcloud.com/ 😉

**Comment by ChristophWurst** [comment]
Please check your browser's developer console for possible js/security errors or warnings and post them here, e.g. with the help of a screenshot. Thx.

**Comment by erudisch** [comment]
OK, here is the output from the developers console. Hope this helps.

![nextcloud_2fa_problem](https://cloud.githubusercontent.com/assets/4243192/25073563/a157e7de-22e9-11e7-9426-285e04cde93d.png)

![nc_2fa_problem_2](https://cloud.githubusercontent.com/assets/4243192/25073592/47f79ec2-22ea-11e7-8d94-d1378a1353b4.png)

Kind Regards
Erik

**Comment by rigrig** [comment]
It looks like it is trying to load javascript files from **w**apps instead of apps.

@erudisch
* Have you installed the app in the `apps` directory, or in a separate `wapps` directory (or possibly moved it later)?
* In your config.php, is there an `apps_paths` entry and if present, what does it look like?
It would look something like this:
```php
'apps_paths' => array(
        array(
                'path'=> '/var/www/nextcloud/apps',
                'url' => '/apps',
                'writable' => true,
        ),
),
```

**Comment by erudisch** [comment]
@rigrig 

Yes, there was an app-path entry in my config.php which looks like this:

`
'apps_paths' =>
    array (
    0 => 
    array (
      'path' => '/usr/share/webapps/nextcloud/apps',
      'url' => '/apps',
      'writable' => false,
    ),
    1 => 
    array (
      'path' => '/var/www/nextcloud/apps',
      'url' => '/wapps',
      'writable' => true,
    ),
  ),
`
So, I removed the second entry and set the first one to 'writable' => true. Now the checkbox for enabling is there and I will activate the 2fa this evening and respond if it works.

I switched from owncloud to nextcloud and have had some problems with it. The solution was to add the app-path entry listed above...

Thnks all for your help!

Very best regards
Erik
